### ISSUE TITLE
Using the example native-modules in the documentation is an error

### ISSUE THREAD

**Issue by chrisglein** [microsoft/react-native-windows#11449]
### Discussed in https://github.com/microsoft/react-native-windows/discussions/11447

<div type='discussions-op-text'>

<sup>Originally posted by **FengZi-lv** April  4, 2023</sup>
When I use the example from https://microsoft.github.io/react-native-windows/docs/native-modules I get the error

# Problem reproduction

> If you need the full code, please check https://github.com/FengZi-lv/ask_it_example

Create a new project
```shell
npx react-native init AskIt
cd AskIt
npx react-native-windows-init --overwrite
```

Add these codes in React Native
`src/lib/NativeFancyMath.ts`
```ts
import type {TurboModule} from 'react-native/Libraries/TurboModule/RCTExport';
import {TurboModuleRegistry} from 'react-native';

export interface Spec extends TurboModule {
  getConstants: () => {
    E: number;
    PI: number;
  };

  add(a: number, b: number): Promise<number>;
}

export default TurboModuleRegistry.get<Spec>('FancyMath') as Spec | null;
```

`index.js`
```js
// /**
//  * @format
//  */
//
// import {AppRegistry} from 'react-native';
// import App from './App';
import {name as appName} from './app.json';
//
// AppRegistry.registerComponent(appName, () => App);

import React, {Component} from 'react';
import {AppRegistry, Alert, Text, View, Button} from 'react-native';

import FancyMath from './src/lib/NativeFancyMath';
import {NativeEventEmitter} from 'react-native';

const FancyMathEventEmitter = new NativeEventEmitter(FancyMath);

class App extends Component {
  eventHandler(result) {
    console.log('Event was fired with: ' + result);
  }

  componentDidMount() {
    // Subscribing to FancyMath.AddEvent
    FancyMathEventEmitter.addListener('AddEvent', this.eventHandler, this);
  }

  componentWillUnmount() {
    // Unsubscribing from FancyMath.AddEvent
    FancyMathEventEmitter.removeListener('AddEvent', this.eventHandler, this);
  }


  _onPressHandler() {
    // Calling FancyMath.add method
    FancyMath.add(
      /* arg a */ FancyMath.Pi,
      /* arg b */ FancyMath.E,
      /* callback */ function (result) {
        Alert.alert(
          'FancyMath',
          `FancyMath says ${FancyMath?.Pi} + ${FancyMath?.E} = ${result}`,
          [{text: 'OK'}],
          {cancelable: false},
        );
      },
    );
  }

  render() {
    return (
      <View>
        <Text>FancyMath says PI = {FancyMath?.Pi}</Text>
        <Text>FancyMath says E = {FancyMath?.E}</Text>
        <Button onPress={this._onPressHandler} title="Click me!"/>
      </View>
    );
  }
}

AppRegistry.registerComponent(appName, () => App);

```

`package.json`
```json
"codegenConfig": {
     "name": "AskIt",
     "type": "modules",
     "jsSrcsDir": "src/lib",
     "windows": {
       "namespace": "AskIt"
     }
   }
```

Add these codes in windows solution
`FancyMath.h`
```cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#pragma once

#include "pch.h"

#include <functional>
#define _USE_MATH_DEFINES
#include <math.h>

#include "NativeModules.h"
#include "../../codegen/NativeFancyMathSpec.g.h"

namespace App
{
    REACT_MODULE(FancyMath);    // 模块名称
    struct FancyMath
    {
        using ModuleSpec = FancyMathSpec;

        REACT_GET_CONSTANTS(GetConstants)   // 返回的常量
            FancyMathSpec_Constants GetConstants() noexcept
        {
            FancyMathSpec_Constants constants;
            constants.E = M_E;
            constants.Pi = M_PI;
            return constants;
        }

        REACT_METHOD(Add, L"add");
        double Add(double a, double b) noexcept
        {
            double result = a + b;
            AddEvent(result);
            return result;
        }

        REACT_EVENT(AddEvent);
        std::function<void(double)> AddEvent;
    };
}

```

`ReactPackageProvider.h`
```cpp
#pragma once

#include "winrt/Microsoft.ReactNative.h"

namespace winrt::AskIt::implementation
{
    struct ReactPackageProvider : winrt::implements<ReactPackageProvider, winrt::Microsoft::ReactNative::IReactPackageProvider>
    {
    public: // IReactPackageProvider
        void CreatePackage(winrt::Microsoft::ReactNative::IReactPackageBuilder const &packageBuilder) noexcept;
    };
} // namespace winrt::AskIt::implementation

namespace winrt::AskIt::factory_implementation
{

    struct ReactPackageProvider : ReactPackageProviderT<ReactPackageProvider, implementation::ReactPackageProvider> {};

} // namespace winrt::AskIt::factory_implementation
```

`ReactPackageProvider.cpp`
```cpp
#include "pch.h"
#include "ReactPackageProvider.h"
#include "NativeModules.h"


// NOTE: You must include the headers of your native modules here in
// order for the AddAttributedModules call below to find them.
#include "FancyMath.h"

using namespace winrt::Microsoft::ReactNative;

namespace winrt::AskIt::implementation
{

void ReactPackageProvider::CreatePackage(IReactPackageBuilder const &packageBuilder) noexcept
{
    AddAttributedModules(packageBuilder, true);
}

} // namespace winrt::AskIt::implementation
```

Run `npx react-native run-windows`

At this point the program throws an error
```log
√ Restoring NuGet packages
 √ Auto-linking...
Success: No auto-linking changes necessary. (24ms)
 √ Found Solution: C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt.sln
 i Build configuration: Debug
 i Build platform: x64
 × Building Solution: C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\XamlCompiler\Microsoft.Windows.UI.X...
 × Build failed with message C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\ReactPackageProvider.h(17,56): error C2504: ��ReactPackageProviderT��: δ������� (����Դ�ļ� ReactPackageProvider.cpp) [C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\AskIt.vcxproj]. Check your build configuration.
Command failed. Re-run the command with --logging for more information.
```

```
vs error message



Severity Code Description Item File Line Suppressed Status
Error MSB4181 The 'CompileXaml' task returned false, but no error was logged. AskIt C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\XamlCompiler\Microsoft.Windows.UI.Xaml.Common.targets 486
Error (active) E1696 Cannot open source file "../../codegen/NativeFancyMathSpec.g.h" AskIt C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\FancyMath.h 13
Error(Activity) E0864 ReactPackageProviderT is not a template AskIt C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\ReactPackageProvider.h 17
Error C2504 "ReactPackageProviderT": undefined base class (compile source file ReactPackageProvider.cpp) AskIt C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\ReactPackageProvider.h 17
Error C2143 Syntax error: missing "," (before "<") (compile source file ReactPackageProvider.cpp) AskIt C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\ReactPackageProvider.h 17
Error C2504 "ReactPackageProviderT": undefined base class (compile source file App.cpp) AskIt C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\ReactPackageProvider.h 17
Error C2143 Syntax error: missing "," (before "<") (compile source file App.cpp) AskIt C:\Users\fengz\Documents\Project\ReactNative\AskIt\windows\AskIt\ReactPackageProvider.h 17
Error C1083 Cannot open include file: “../../codegen/NativeFancyMathSpec.g.h”: No such file or directory (compile source file ReactPackageProvider.cpp) AskIt C:\Users\fengz\Documents\Project\ReactNative\AskIt\ windows\AskIt\FancyMath.h 13
```

I noticed that `./codegen` does not generate the expected `./codegen/NativeFancyMathSpec.g.h`

what should i do, thanks</div>

**Comment by chrisglein** [comment]
@FengZi-lv FYI here's the issue created from the discussion.

This codegen is very new and may not be ready until 0.72. We're currently debating whether the right things to unblock this landed in 0.71.
Did the `./codegen` folder get created?
You can try running the command manually, which is `npx react-native codegen-windows`?
Also, I'm going to convert this from a discussion to an issue since you're trying something that should work and it isn't :)

**Comment by FengZi-lv** [comment]
@chrisglein I'm sorry for taking so long to reply to you. Thank you for your attention to this issue.
I no longer use react native, I no longer pay attention to this issue, and I have forgotten many details.
I didn't solve this problem at the time, I used Flutter instead of React Native now.
